### ISSUE TITLE
How should multi-controlled gates be added?

### ISSUE THREAD

**Issue by Cryoris** [Qiskit/qiskit-terra#4451]
Multi-controlled gates are currently partially objects (`MCX MCU1`) and partially functions monkey-patched to the circuit (`mcr*`).  Some, as the `MCX` are moving to the circuit library, some are already, as the `GMS`. In principle library objects can be added to the circuit using `compose`, but all multi-controlled gates right now are accessible as circuit methods.

This raises two questions:
1. Which multi-controlled gates should have their proper class? 
2. How should they be added to circuits?

Possible solution:
1. Only the ones where we know of multiple implementations, such as `MCX`.
2. Two options:
a. Don't allow circuit methods:
    ```python
    circuit.append(RXGate(0.2).control(5), [0, 1, 2, 3, 4, 5])
    circuit.compose(MCX(5))
    ```
    b. Allow the controlled circuit methods to accept lists of qubits:
    ```python
    circuit.crx(0.2, [0, 1, 2, 3, 4], 5)  # goes and does RXGate(0.2).control(5)
    circuit.cx([0, 1, 2, 3, 4], 5)  # goes and does MCX(5)
    ```
    maybe the `cx` is still qualified to have `mcx` to also allow arguments such as `mode`. 
    Advantage: easy adding of multi-controlled gates, no additional methods
    Problem: `circuit.cx([0,1], 2)` currently means apply two `CX` gates.

    c. Keep `mcrx` as circuit method and 
    ```python
    circuit.mcrx(0.2, [0, 1, 2, 3, 4], 5)  # goes and does RXGate(0.2).control(5)
    circuit.mcx([0, 1, 2, 3, 4], 5)  # goes and does MCX(5)
    ```
    Advantage: easy adding of multi-controlled gates,
    But: additional methods

I'd prefer one of the 2b and 2c, since it gives us the power of multi-controlled gates with an intuitive and somewhat expected syntax. Also we don't have to add dummy `MCGate` classes that just do Qiskit's default `Gate.control` under the hood and don't add any value to the library.

Since 2b has problematic syntax, 2c seems a good solution.

Are there opinions that the multi-controlled gates should only be library objects and not offer circuit methods? (@ajavadia @jaygambetta)

**Comment by nonhermitian** [comment]
>  I'm not aware of a use-case for the current interpretation where cx([0, 1], 2) constructs two CX gates

This is actually quite nice to have, and saves writing a for-loop just to add gates where you know they are going to go.  Useful for cluster states, GHZ states etc.

**Comment by ajavadia** [comment]
@nonhermitian you are right but I think an internal poll indicated that most people expected this to create a multi-controlled gate.
